src/Geralt.Tests/XChaCha20Poly1305Tests.cs

```diff
@@ -183,4 +183,208 @@ public void Decrypt_InvalidKey()
+    [TestMethod]
+    public void IncrementalEncryptDecrypt_Chunked_ValidInputs()
+    {
+        Span<byte> header = stackalloc byte[IncrementalXChaCha20Poly1305.HeaderBytes];
+        int plaintext1Length = 50;
+        int plaintext2Length = 50;
+        int plaintext3Length = Plaintext.Length - plaintext1Length - plaintext2Length;
+        Span<byte> plaintext = stackalloc byte[Plaintext.Length];
+        Span<byte> ciphertext1 = stackalloc byte[plaintext1Length + IncrementalXChaCha20Poly1305.TagSize];
+        Span<byte> ciphertext2 = stackalloc byte[plaintext2Length + IncrementalXChaCha20Poly1305.TagSize];
+        Span<byte> ciphertext3 = stackalloc byte[plaintext3Length + IncrementalXChaCha20Poly1305.TagSize];
+        byte outTag = 0xFF;
+        using var encryptor = new IncrementalXChaCha20Poly1305(false, header, Key);
+        encryptor.Push(ciphertext1, Plaintext[0..plaintext1Length], IncrementalXChaCha20Poly1305.TagMessage);
+        encryptor.Push(ciphertext2, Plaintext[plaintext1Length..(plaintext1Length + plaintext2Length)], IncrementalXChaCha20Poly1305.TagMessage);
+        encryptor.Push(ciphertext3, Plaintext[(plaintext1Length + plaintext2Length)..Plaintext.Length], IncrementalXChaCha20Poly1305.TagFinal);
+        using var decryptor = new IncrementalXChaCha20Poly1305(true, header, Key);
+        decryptor.Pull(plaintext[0..plaintext1Length], ciphertext1, tag: ref outTag);
+        Assert.AreEqual(IncrementalXChaCha20Poly1305.TagMessage, outTag);
+        decryptor.Pull(plaintext[plaintext1Length..(plaintext1Length + plaintext2Length)], ciphertext2, tag: ref outTag);
+        Assert.AreEqual(IncrementalXChaCha20Poly1305.TagMessage, outTag);
+        decryptor.Pull(plaintext[(plaintext1Length + plaintext2Length)..Plaintext.Length], ciphertext3, tag: ref outTag);
+        Assert.AreEqual(IncrementalXChaCha20Poly1305.TagFinal, outTag);
+        Assert.IsTrue(plaintext.SequenceEqual(Plaintext));
+    }
+
+    [TestMethod]
+    public void IncrementalEncryptDecrypt_ValidInputs_WithAssociatedData()
+    {
+        Span<byte> header = stackalloc byte[IncrementalXChaCha20Poly1305.HeaderBytes];
+        Span<byte> plaintext = stackalloc byte[Plaintext.Length];
+        Span<byte> ciphertext = stackalloc byte[Plaintext.Length + IncrementalXChaCha20Poly1305.TagSize];
+        byte[] associatedData = new byte[] { 0x01, 0x02, 0x03, 0x04 };
+        byte outTag = 0xFF;
+        using var encryptor = new IncrementalXChaCha20Poly1305(false, header, Key);
+        encryptor.Push(ciphertext, Plaintext, associatedData, IncrementalXChaCha20Poly1305.TagFinal);
+        using var decryptor = new IncrementalXChaCha20Poly1305(true, header, Key);
+        decryptor.Pull(plaintext, ciphertext, associatedData, tag: ref outTag);
+        Assert.AreEqual(IncrementalXChaCha20Poly1305.TagFinal, outTag);
+        Assert.IsTrue(plaintext.SequenceEqual(Plaintext));
+    }
+
+    [TestMethod]
+    public void IncrementalEncryptDecrypt_ValidInputs_WithReKey()

```

Comment:
We can probably get away with merging `Rekey()` into the currently non-rekeying test to save some lines.

Reply:
I prefer keeping this test.  If this test fails we know the issue has something to do with rekey and not something else.